dev_docs/POLICIES.md

```diff
@@ -0,0 +1,61 @@
+# TileDB-SOMA Development Policies
+
+## Version policy
+
+TileDB-SOMA uses a loose variation of [semantic versioning](https://semver.org/) to organize and govern API compatibility, deprecations and version numbers.
+
+TileDB-SOMA also uses lifecycle tags to indicate the maturity of an interface. All public API should have a lifecycle tag in docstrings. These tags are patterned after the RStudio lifecycle stage model. Tags are:
+
+- ``experimental``: Under active development and may undergo significant and breaking changes.
+- ``maturing``: Under active development but the interface and behavior have stabilized and are unlikely to change significantly but breaking changes are still possible.
+- ``stable``: The interface is considered stable and breaking changes will be avoided where possible. Breaking changes that cannot be avoided will be accompanied by a major version bump.
+- ``deprecated``: The API is no longer recommended for use and may be removed in a future release.
```

Comment:
```suggestion
- ``deprecated``: The API is no longer recommended for use and will be removed in a future release.
```
The numpy documentation had a good point about not deprecating things that aren't going to be removed.